providers/chilipiper/url.go

```diff
@@ -1,9 +1,44 @@
 package chilipiper
 
-import "github.com/amp-labs/connectors/common/urlbuilder"
+import (
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
 
 const restAPIVersionPrefix = "api/fire-edge/v1/org"
 
-func (conn *Connector) buildReadURL(object string) (*urlbuilder.URL, error) {
-	return urlbuilder.New(conn.BaseURL, restAPIVersionPrefix, object)
+func (conn *Connector) buildReadURL(config common.ReadParams) (*urlbuilder.URL, error) {
```

Comment:
minor: buildURL can have a page size as a parameter, and we can reduce these two into one function that can be used after checking for a next page.